From 6386250066ff7903e1961935d64f0ad1307a79f0 Mon Sep 17 00:00:00 2001
From: Hauke Mehrtens <hauke.mehrtens@intel.com>
Date: Wed, 5 Jun 2019 11:17:45 +0200
Subject: [PATCH] PONRTSYS-4292: Fix setting ptp clock rate

The clk_round_rate() function does not modify the hardware, it only
returns the nearest clock rate. Here we want to modify the clock rate,
so set it and read it back.
---
 drivers/net/ethernet/lantiq/switch-api/mac/mac_drv.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/drivers/net/ethernet/lantiq/switch-api/mac/mac_drv.c b/drivers/net/ethernet/lantiq/switch-api/mac/mac_drv.c
index f5a52906774d..e760cd17b6ae 100644
--- a/drivers/net/ethernet/lantiq/switch-api/mac/mac_drv.c
+++ b/drivers/net/ethernet/lantiq/switch-api/mac/mac_drv.c
@@ -200,12 +200,14 @@ static int mac_probe(struct platform_device *pdev)
 		return 0;
 	}
 
-	ret = clk_round_rate(pdata->ker_ptp_clk, pdata->ptp_clk);
-	if (ret < 0) {
-		dev_err(dev, "Failed to set MAC %d ptp clock to %i Hz\n",
-			pdata->mac_idx, pdata->ptp_clk);
+	ret = clk_set_rate(pdata->ker_ptp_clk, pdata->ptp_clk);
+	if (ret) {
+		dev_err(dev, "Failed to ptp clock to %i Hz: err: %i\n",
+			pdata->ptp_clk, ret);
 		return ret;
 	}
+
+	ret = clk_get_rate(pdata->ker_ptp_clk);
 	if (ret != pdata->ptp_clk) {
 		dev_err(dev, "Expected PTP clock rate unsupported, will use %i Hz instead\n",
 			ret);
